3DAmsterdam/Assets/Netherlands3D/Scripts/Traffic/Car.cs

```diff
@@ -37,10 +50,64 @@ void Start()
                 car.SetActive(false);
             }
             // Chooses a random car out of all the car objects.
-            cars[Random.Range(0, cars.Length - 1)].SetActive(true);
+
+            carType = cars[UnityEngine.Random.Range(0, cars.Length)];
+            vehicleProperties = carType.GetComponent<VehicleProperties>();
+            if (carType.name == "BasicCar")
+            {
+                ApplySettings(carType);
+            }
+            carType.SetActive(true);
             speed = TrafficSimulator.Instance.carSpeed;
         }
 
+        private void ApplySettings(GameObject car)
+        {
+            float colorPercentage = UnityEngine.Random.Range(0.0f, 1.0f);
+            car.GetComponent<Renderer>().material.color = GenerateColor(colorPercentage);
+        }
+
+        private Color32 GenerateColor(float colorPercentage)
+        {
+            if(colorPercentage > 0.66f)
+            {
+                return new Color32(145, 145, 145, 255); //gray 34%
+            }
+            if(colorPercentage <= 0.66f && colorPercentage > 0.42f)
+            {
+                return new Color32(30, 30, 30, 255); //black 24%
+            }
+            if(colorPercentage <= 0.42f && colorPercentage > 0.28f)
+            {
+                return new Color32(57, 71, 245, 255); //blue 14%
+            }
+            if(colorPercentage <= 0.28f && colorPercentage > 0.15f)
+            {
+                return new Color32(220, 220, 220, 255); //white 13%
+            }
+            if(colorPercentage <= 0.15f && colorPercentage > 0.08f)
+            {
+                return new Color32(200, 10, 10, 255); //red 7%
+            } 
+            if(colorPercentage <= 0.08f && colorPercentage > 0.05f)
+            {
+                return new Color32(10, 160, 10, 255); //green 3%
+            }  
+            if(colorPercentage <= 0.05f && colorPercentage > 0.03f)
+            {
+                return new Color32(100, 35, 15, 255); //brown 2%
+            }    
+            if(colorPercentage <= 0.03f && colorPercentage > 0.02f)
+            {
+                return new Color32(244, 226,198, 255); //beige 1%
+            } 
+            if(colorPercentage <= 0.02f && colorPercentage > 0.01f)
+            {
+                return new Color32(255, 223, 0, 255); //yellow 1%
+            }  
```

Comment:
these can all be 'else if''s.
ps. this is a nice case to make/use a 'loot table'. there are a lot of nice ways to make those, and will also be very usefull for your games. The dropchance of a legendary sword would in this case be the selection of a specific rar color for a car ;)

Reply:
Never thought of looking up how those are made, seems obvious for this use case, thanks!